galaxyxml/tool/parameters/__init__.py

```diff
@@ -69,14 +73,87 @@ def __init__(
         super(Stdio, self).__init__(**params)
 
 
+class Macros(XMLParam):
+    name = "macros"
+
+    def acceptable_child(self, child):
+        return isinstance(child, Macro) \
+               or isinstance(child, Import)
+
+
+class Macro(XMLParam):
+    name = "xml"
+
+    def __init__(self, name):
+        params = Util.clean_kwargs(locals().copy())
+        passed_kwargs = {}
+        passed_kwargs['name'] = params['name']
+        super(Macro, self).__init__(**passed_kwargs)
+
+    def acceptable_child(self, child):
+        return issubclass(type(child), XMLParam) \
```

Comment:
Why `type(child)`?

Reply:
that's a good question. I can't say I have a good answer. Probably just remove it.

Reply:
Actually its correct because `issubclass` requires a class. I mixed this up with `isinstance` which is also used in the `acceptable_child` functions.